### PR TITLE
Merge gen2 release to iam auth

### PR DESCRIPTION
**Pull request by sobolk**
#### Description of changes

This PR contains 2 changes:
1. It merges `feature/gen2-release` to `feature/iam-auth`
2. It splits `sql-models.test.ts` into two : `sql-models.test.ts` which stays in parity with base branch and new `sql-iam-access.test.ts` which contains scenarios related to IAM access only. The reasons for this are: 1. these tests focus on independent aspects of sql data base, 2. they evolved independently in both branches into unresolvable merge.

**Note to reviewers: I have commented on files that were refactored post merge. Please focus on them. Uncommented files were not modified post merge.**

#### Description of how you validated changes

PR checks & e2e workflow.

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] Any CDK or CloudFormation parameter changes are called out explicitly

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: codebuild_specs/e2e_workflow.yml [hunk 1]
```diff
@@ -1020,7 +1020,7 @@ batch:
         compute-type: BUILD_GENERAL1_SMALL
         variables:
           TEST_SUITE: src/__tests__/schema-searchable.test.ts
-          CLI_REGION: eu-north-1
```
Comment (sobolk) [inline codebuild_specs/e2e_workflow.yml:1023]: This file has been regenerated post merge after splitting the test.

File: dependency_licenses.txt [hunk 2]
```diff
@@ -2176,7 +2176,7 @@ Apache License
...
...s-v3.git (@aws-sdk/service-error-classification), https://github.com/aws/aws-sdk-js-v3.git (@aws-sdk/url-parser), https://github.com/aws/aws-sdk-js-v3.git (@aws-sdk/url-parser-native), https://github.com/aws/aws-sdk-js-v3.git (@aws-sdk/util-defaults-mode-browser), https://github.com/aws/aws-sdk-js-v3.git (@aws-sdk/util-defaults-mode-node), https://github.com/aws/aws-sdk-js-v3.git (@aws-sdk/util-user-agent-browser), https://github.com/aws/aws-sdk-js-v3.git (@aws-sdk/util-user-agent-node), https://github.com/awslabs/smithy-typescript.git (@smithy/eventstream-codec), https://github.com/awslabs/smithy-typescript.git (@smithy/middleware-retry), https://github.com/awslabs/smithy-typescript.git (@smithy/querystring-builder), https://github.com/awslabs/smithy-typescript.git (@smithy/querystring-parser), https://github.com/awslabs/smithy-typescript.git (@smithy/service-error-classification), https://github.com/awslabs/smithy-typescript.git (@smithy/url-parser), https://github.com/awslabs/smithy-typescript.git (@smithy/util-defaults-mode-browser), https://github.com/awslabs/smithy-typescript.git (@smithy/util-defaults-mode-node). This software contains the following license and notice below:
```
Comment (sobolk) [inline dependency_licenses.txt:2179]: This file was regenerated due to merge conflict.

File: packages/amplify-e2e-core/src/utils/rds.ts [hunk 3]
```diff
@@ -26,12 +28,12 @@ const DEFAULT_SECURITY_GROUP = 'default';
 const IPIFY_URL = 'https://api.ipify.org/';
 const AWSCHECKIP_URL = 'https://checkip.amazonaws.com/';
 
-type RDSConfig = {
+export type RDSConfig = {
```
Comment (sobolk) [inline packages/amplify-e2e-core/src/utils/rds.ts:31]: There's one extra change here.
I added `export` keyword here to make this available to callers.
(exported functions below use this to type inputs, it was forgotten export that surfaced during refactor of construct tests)

File: packages/amplify-graphql-api-construct-tests/src/__tests__/backends/sql-iam-access/app.ts [hunk 4]
```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env node
```
Comment (sobolk) [inline packages/amplify-graphql-api-construct-tests/src/__tests__/backends/sql-iam-access/app.ts:1]: This file is result of splitting the `sql-models.test.ts`.

This file was duplicated from `packages/amplify-graphql-api-construct-tests/src/__tests__/backends/sql-models/app.ts` and then unused pieces were removed.

File: packages/amplify-graphql-api-construct-tests/src/__tests__/backends/sql-models/app.ts [hunk 5]
```diff
@@ -1,32 +1,26 @@
 #!/usr/bin/env node
 import 'source-map-support/register';
-import { App, Stack, Duration, CfnOutput, RemovalPolicy } from 'aws-cdk-lib';
+import { App, Stack, Duration, CfnOutput } from 'aws-cdk-lib';
```
Comment (sobolk) [inline packages/amplify-graphql-api-construct-tests/src/__tests__/backends/sql-models/app.ts:3]: This file is result of splitting the `sql-models.test.ts`.

Changes in this file are to reset it's state to match `feature/gen2-release` branch.

File: packages/amplify-graphql-api-construct-tests/src/__tests__/sql-iam-access.test.ts [hunk 6]
```diff
@@ -0,0 +1,337 @@
+import * as path from 'path';
```
Comment (sobolk) [inline packages/amplify-graphql-api-construct-tests/src/__tests__/sql-iam-access.test.ts:1]: This file is result of splitting the `sql-models.test.ts`.

This file is result of moving "new iam access tests" outside of `sql-models.test.ts`.

File: packages/amplify-graphql-api-construct-tests/src/sql-datatabase-controller.ts [hunk 7]
```diff
@@ -0,0 +1,172 @@
+import path from 'path';
```
Comment (sobolk) [inline packages/amplify-graphql-api-construct-tests/src/sql-datatabase-controller.ts:1]: This file is result of splitting the `sql-models.test.ts`.

While splitting tests, I extracted common "db setup - tear down, etc" code into this class. and changed both tests to use it.

Logic is copied pretty much verbatim from tests, but packaged in a class in order to encapsulate workflow.

**Review by sundersc — APPROVED** [review]

**Review by phani-srikar — APPROVED** [review]

**Review by palpatim — APPROVED** [review]

**Pull request merged**